ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```diff
@@ -156,6 +159,10 @@ private void exportFilesAsList() throws SemanticException, IOException, LoginExc
         }
         done = true;
       } catch (IOException e) {
+        if (e instanceof FileNotFoundException) {
+          logger.error("exporting data files in dir : " + dataPathList + " to " + exportRootDataDir + " failed");
+          throw new FileNotFoundException(FILE_NOT_FOUND.format(e.getMessage()));
```

Comment:
For same use-case with Acid tables where table dir is missing throws IOException from Utils.getDataPathList instead of FileNotFoundException. Also, add a test case to validate this.

Reply:
hadoop team has confirmed that they throw file not found for all cases where file/dir is missing.